src/detectors/node/NodeDetector.ts

```diff
@@ -0,0 +1,184 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { join } from 'path';
+import { AzExtFsExtra, parseError } from "vscode-azureextensionui";
+import { parse } from 'yaml';
+import { NodeConstants } from "./nodeConstants";
+
+type PackageJson = {
+    engines?: {
+        node?: string
+    },
+    devDependencies?: {
+        [key: string]: string
+    },
+    dependencies?: {
+        [key: string]: string
+    }
+}
+
+type FrameworkInfo = {
+    framework: string,
+    version: string
+}
+
+export type DetectorResults = {
+    platform: string | undefined,
+    platformVersion: string | undefined,
+    appDirectory: string | undefined,
+    frameworks: FrameworkInfo[],
+    hasLernaJsonFile: boolean,
+    hasLageConfigJSFile: boolean,
+    lernaNpmClient: string | undefined,
+    hasYarnrcYmlFile: boolean,
+    isYarnLockFileValidYamlFormat: boolean,
+}
+
+
+export class NodeDetector {
```

Comment:
A comment somewhere prominent in this file explaining that this code is modeled after Oryx, and a link to the Oryx detector would be good.

And now that I think about it, probably the NodeConstants.ts file also.